File: api/Portfolio.Domain/Models/Skill.cs
using Portfolio.Domain.Models.Common;
using System.Collections.Generic;

namespace Portfolio.Domain.Models
{
    public class Skill : BaseEntity, IHasDisplayNumber
    {
        #region Properties

        public string Name { get; set; }

        public string IconPath { get; set; }

        public int DisplayNumber { get; set; }

        public int SkillGroupId { get; set; }

        public SkillGroup SkillGroup { get; set; }

        public ICollection<Project> Projects { get; set; }

        #endregion
    }
}

File: admin/src/app/data/Skill.ts
import { SkillGroup } from './SkillGroup';

export enum SkillType {
    FrontEnd,
    BackEnd,
    Other
}
   
export interface Skill {
    name: string,
    iconPath: string,
    displayNumber: number,
    skillGroupId: number,
    skillGroup: SkillGroup,
    id: number
}

export interface CreateUpdateSkill {
    name: string,
    iconPath: string,
    displayNumber: number,
    skillGroupId: number,
    id: number
}
   